engine/src/main/scala/cromwell/webservice/CromwellApiService.scala

```diff
@@ -285,16 +290,40 @@ object CromwellApiService {
   import spray.json._
 
   implicit class EnhancedThrowable(val e: Throwable) extends AnyVal {
-    def failRequest(statusCode: StatusCode): Route = complete((statusCode, APIResponse.fail(e).toJson.prettyPrint))
-    def errorRequest(statusCode: StatusCode): Route = complete((statusCode, APIResponse.error(e).toJson.prettyPrint))
+    def failRequest(statusCode: StatusCode, warnings: Seq[String] = Vector.empty): Route = {
+      completeResponse(statusCode, APIResponse.fail(e).toJson.prettyPrint, warnings)
+    }
+    def errorRequest(statusCode: StatusCode, warnings: Seq[String] = Vector.empty): Route = {
+      completeResponse(statusCode, APIResponse.error(e).toJson.prettyPrint, warnings)
+    }
+  }
+
+  def completeResponse[A](statusCode: StatusCode, value: A, warnings: Seq[String])
+                         (implicit mt: ToEntityMarshaller[A]): Route = {
+    val warningHeaders = warnings.toIndexedSeq map { warning =>
+      /*
+      Need a quoted string.
+      https://stackoverflow.com/questions/7886782
+
+      Using a poor version of ~~#!
+      https://github.com/akka/akka-http/blob/v10.0.9/akka-http-core/src/main/scala/akka/http/impl/util/Rendering.scala#L206
+       */
+      val quotedString = "\"" + warning.replaceAll("\"","\\\\\"").replaceAll("[\\r\\n]+", " ").trim + "\""
```

Comment:
Whoa! here be { 🐲 🐉 🐲 }!

"I'm going to escape my own HTML" sounds like a bad idea if this is a security mechanism. If it's just "so it's not a horrible UI" then I'm less worried...

Reply:
It's not security, and not HTML. It's HTTP. HTTP has some weird type of `quoted-string` type that goes in the headers. It's is poorly defined IMHO, and I really wish someone else would have implemented this so I could look at what they're doing. The closest I could find was the cryptically named `~~#!` operator buried privately in akka-http-core.

See the links in the comments for more info, or if more comments would be better, let me know.

Reply:
Is there any reason you can't use the existing `~~#!` method?

Reply:
All of `akka.http.impl.util.Rendering` is package-private. In the past I've used runtime-reflection to "fix" private access methods. But that method didn't strip the newlines we're adding. So I went with just adding a reference pointer showing that this wasn't the first method to handle this weird (again imho) HTTP quoted-string.

Reply:
This isn't a seagull request or anything, but since my eyebrow was literally just raised I figured I should throw it out there. How is it that we're running into something which apparently no one else does if it requires hijinks of this caliber? 

Is this a case where technically it's supposed to work a certain way but most people don't bother? Is it that typically people aren't putting quoted strings in that header? Something else?

Reply:
> How is it that we're running into something which apparently no one else does if it requires hijinks of this caliber?

Here is _one_ example of someone else jumping through loops to return a proper Warning 299: https://github.com/elastic/elasticsearch/blob/v5.5.2/core/src/main/java/org/elasticsearch/common/logging/DeprecationLogger.java#L307-L315

There are more examples even in this file of returning proper deprecation warnings over HTTP. Oracle Java and Akka Http don't have a library to help with this header, afaik. Maybe Apache does?

Overall, I get the impression that this was a well intentioned spec that needed a [second round](https://stackoverflow.com/questions/7886782/what-is-the-exact-syntax-and-semantics-of-a-quoted-string-in-the-http1-1-rfc2616#7888497) of cleanup.

> Is this a case where technically it's supposed to work a certain way but most people don't bother?

Most people (ex: see rest of code in ES class) are returning a very basic "go fix this" warning messages to their clients. Our warnings originate as a paragraph including line feeds. Not horrible, but requires some extra stripping of line feeds before it's sent over HTTP, where a CRLF is also defined as the official separator for header fields.

Reply:
mentioned this at standup but this wasn't a request for action. i just wanted to understand what was happening